uefi/src/lib.rs

```diff
@@ -118,6 +118,25 @@
 //! Contributions in form of a PR are also highly welcome. Check our
 //! [contributing guide][contributing] for details.
 //!
+//! # Comparison to other Projects in the Ecosystem
+//!
+//! ## Rust `std` implementation
+//!
+//! There is an ongoing effort for a `std` implementation of the Rust standard
+//! library. ([Platform description][rustc-uefi-std],
+//! [tracking issue][uefi-std-tr-issue]), that is yet far from being mature. As
+//! of Mid-2024, we recommend to use our `uefi` library in a `no_std` binary to
+//! create EFI images.
+//!
+//! We will closely monitor the situation and update the documentation
+//! accordingly, once the `std` implementation is more mature.
```

Comment:
IMO we should not pass a judgement here on whether it's mature (parts are perfectly usable, parts are still in development, and it really just depends on what you need out of it).

I would drop everything from "that is yet far from being mature" on. Instead, just give the facts: explain that it allows you to write UEFI programs that look very similar to normal programs running on top of an OS, and also mention that it is still under development (without promising that we'll closely monitor it, since I think that monitoring is better done by potential users of it)